### PR TITLE
Add escape hatch for Series merging

### PR DESCRIPTION
**Pull request by revam**
**Changes**

This is an universal solution for plugins to override how series are merged.
The reason to override is so we can set the same provider id on multiple items without merging them, while using another id for merging them. Having an (optional) provider id not tied to any online database allows plugins to use their own rules for merging series.

**Issues**

Follow up PR for #7497

### CODE REVIEW

**Comment by revam** [comment]
I don't know if it should show up in the UI or not, but I can add another commit to add the IExternalId needed if so desired.

**Comment by revam** [comment]
Also, this doesn't change — or try to reuse — any existing logic, as opposed to what my previous attempt at solving this problem did (see the linked PR for details).

File: MediaBrowser.Model/Entities/MetadataProvider.cs [hunk 1]
```diff
@@ -7,6 +7,12 @@ namespace MediaBrowser.Model.Entities
     /// </summary>
     public enum MetadataProvider
     {
+        /// <summary>
+        /// This provider is an escape hatch for the user to override the series
+        /// merging.
+        /// </summary>
+        User = 0,
```
Comment (cvium) [inline MediaBrowser.Model/Entities/MetadataProvider.cs:14]: Wouldn't it be better to call it `Custom`?

Comment (revam) [reply MediaBrowser.Model/Entities/MetadataProvider.cs:14]: That wpuød probably be better. Will redo it later today

Comment (cvium) [reply MediaBrowser.Model/Entities/MetadataProvider.cs:14]: And the xmldoc is also a bit too Series specific. It can technically be used by all entities.

Comment (revam) [reply MediaBrowser.Model/Entities/MetadataProvider.cs]: Updated the enum name and added a more generic description.

**Comment by revam** [comment]
Is there anything more that should be done, or is there anything else holding back this PR from being merged? @cvium

**Comment by revam** [comment]
Still waiting for a reply...

**Comment by cvium** [comment]
We are not including further features in 10.8. You should rebase on master.

File: MediaBrowser.Model/Entities/MetadataProvider.cs [hunk 2]
```diff
@@ -7,6 +7,12 @@ namespace MediaBrowser.Model.Entities
     /// </summary>
     public enum MetadataProvider
     {
+        /// <summary>
+        /// This metadata provider is for users and/or plugins to override the
+        /// default merging behaviour.
+        /// </summary>
+        Custom = 0,
```
Comment (cvium) [inline MediaBrowser.Model/Entities/MetadataProvider.cs:14]: Are we certain that 0 is unused? Could be 999 or something instead to avoid collisions

Comment (revam) [reply MediaBrowser.Model/Entities/MetadataProvider.cs:14]: I looked at the hit blame history, and only '1' was previously used.

Comment (revam) [reply MediaBrowser.Model/Entities/MetadataProvider.cs:14]: *git

Comment (revam) [reply MediaBrowser.Model/Entities/MetadataProvider.cs:14]: I can change it to either '20' (the next unused number) or '999' (your suggestion) if you don't want to use '0'

Comment (cvium) [reply MediaBrowser.Model/Entities/MetadataProvider.cs:14]: It's fine

**Review by cvium — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Comment by revam** [comment]
Rebased on latest master since the jobs failed (according to my mail notification)

**Pull request merged**